optax/_src/alias.py

```diff
@@ -189,6 +189,7 @@ def adam(
     b2: float = 0.999,
     eps: float = 1e-8,
     eps_root: float = 0.0,
+    amsgrad: bool = False,
```

Comment:
Let's introduce a separate alias `amsgrad` for this rather than adding a flag to Adam. My thinking is that there will be other improvements/modifications to Adam in the future and we should avoid an accumulation of options in the simple adam setup. Furthermore, I think it would make it easier for user to find `amsgrad` in optax and more obvious in the code that amsgrad is being used.